packages/cart/src/repositories/address.ts

```diff
@@ -0,0 +1,25 @@
+import { Context } from "@medusajs/types"
+import { DALUtils } from "@medusajs/utils"
+import { SqlEntityManager } from "@mikro-orm/postgresql"
+import { Address } from "@models"
+import { CreateAddressDTO, UpdateAddressDTO } from "../types"
+
+export class AddressRepository extends DALUtils.mikroOrmBaseRepositoryFactory<
+  Address,
+  {
+    create: CreateAddressDTO
+  }
+>(Address) {
+  async update(
+    data: { address: Address; update: UpdateAddressDTO }[],
+    context: Context = {}
+  ): Promise<Address[]> {
+    const manager = this.getActiveManager<SqlEntityManager>(context)
+
+    const entities = data.map(({ address, update }) => {
+      return manager.assign(address, update)
+    })
+
+    return await super.update(entities, context)
```

Comment:
todo: I would suggest to assign the data manually and call super, or do not call super and persist here. Cause if something evolve and we end up having the same issue of reference than with the cart then it will be hard to find again ahah wdyt?